adagrams/game.py

```diff
@@ -1,11 +1,132 @@
+# from tests.test_wave_01 import LETTER_POOL
+import random
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+
+SCORE_CHART = {
+    ("A", "E", "I", "O", "U", "L", "N", "R", "S", "T") : 1,
+    ("D", "G") : 2,
+    ("B", "C", "M", "P") : 3,
+    ("F", "H", "V", "W", "Y") : 4,
+    ("K") : 5,
+    ("J","X") : 8,
+    ("Q","Z") : 10,
+}
+
+
 def draw_letters():
-    pass
+    '''
+    Creates a new randomized list of letters from LETTER_POOL.
+    Returns: A randomized list of ten letters.
+    '''
+
+    letters = [letter for letter, letter_frequency in LETTER_POOL.items() for number in range(letter_frequency)]
+
+    random.shuffle(letters)
+
+    return letters[:10]
 
 def uses_available_letters(word, letter_bank):
-    pass
+    '''
+    Checks that each letter is listed in the dictionary as >= 
+    as used in the word.
+
+    Parameters: string, dict (word and letter frequency)
+    Returns a Boolean value representing the validity of the input.
+    '''
+
+    word_uppercase = word.upper()
+    
+    for letter in word_uppercase:
+        if letter not in letter_bank or (word_uppercase.count(letter) > letter_bank.count(letter)):
+            return False
+        
+    return True
+
 
 def score_word(word):
-    pass
+    '''
+    Score is given to each letter based on the score chart dictionary.
+    If the word is between 7 and 10, add 8 points to score.
+
+    Parameters: A string of the word.
+
+    Returns the score of the word.
+    '''
+    score = 0
+    word_uppercase = word.upper()
+    if len(word_uppercase) >= 7 and len(word_uppercase) <= 10: 
```

Comment:
We could rewrite this to call `len` once with chained comparison operators:
```suggestion
    if 7 <= len(word_uppercase) <= 10: 
```
Some resources on chaining comparison operators if y'all are interested:
https://docs.python.org/3/reference/expressions.html#comparisons
https://www.geeksforgeeks.org/chaining-comparison-operators-python/

Reply:
Hey, Kelsey! I was actually really struggling with this one. I initially wrote it exactly as you suggested, but it wasn't working. I still can't figure out why.

Reply:
Was there an error, or did the logic produce the wrong result? I'd be curious what you see stepping through with the debugger.